orttraining/orttraining/training_ops/cpu/aten_ops/aten_op_config.h

```diff
@@ -121,6 +131,21 @@ struct ATenOperatorConfig {
 
     return has_default_value;
   }
+
+  bool IsInputOnCpu(size_t input_index, bool is_backward) const {
+    const auto& argument_configs = is_backward ? backward_argument_configs : forward_argument_configs;
+    size_t tensor_input_index = 0;
+    for (const auto& argument_config : argument_configs) {
```

Comment:
why need a loop here? 
can we look up by argument_configs[input_index] ?

Reply:
We need this loop. The argument input_index means the input_index_th node input, which includes tensor inputs only. But the argument_configs contains both tensor and non-tensor arguments for ATen function. If some non-tensor arguments are before tensor arguemnts, the input_index cannot map to the argument_configs of same position.

Actually after the discussion with exporter that day, we agree that to move all non-tensor arguments (attributes for now) to CPU input tensor inputs. Once we have this, we can get rid of the loop.